chrome-extension/background/utils/processSteps.js

```diff
@@ -0,0 +1,98 @@
+/*global chrome*/
+export const processSteps = async (data) => {
+  const steps = data.steps;
+  const server_url = "http://localhost:8000"; // Cant have env variables in chrome extensions
+
+  const active_tab = await chrome.tabs.query({
+    active: true,
+    currentWindow: true,
+  });
+
+  await chrome.scripting.executeScript({
+    target: { tabId: active_tab[0].id },
+    files: ["background/utils/executeActionsScript.js"],
+  });
+
+  let init_time = null;
+  for (const step of steps) {
+    init_time = new Date().getTime();
+    step.state = "in-progress";
+
+    const image = await chrome.tabs.captureVisibleTab();
+    const data = {
+      instruction: {
+        ...step.instruction,
+      },
+      currentView: image,
+    };
+
+    const call = await fetch(`${server_url}/instruction`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(data),
+    });
+
+    const action_response = await call.json();
+
+    await chrome.scripting.executeScript({
+      target: { tabId: active_tab[0].id },
+      args: [{ action_response }],
+      func: (vars) => Object.assign(self, vars),
+    });
+
+    const action_execution_response = await chrome.scripting
+      .executeScript({
+        target: { tabId: active_tab[0].id },
+        // eslint-disable-next-line no-undef -- executeAction is defined in the injected script
+        func: () => executeAction(),
+      })
+      .then((response) => JSON.parse(response[0].result));
```

Comment:
esto no se puede simplificar si `executeAction` recibe `action_response`? y así ya no se necesita el `self`.

```suggestion
    const action_execution_response = await chrome.scripting
      .executeScript({
        target: { tabId: active_tab[0].id },
        // eslint-disable-next-line no-undef -- executeAction is defined in the injected script
        func: () => executeAction({ action_response }),
      })
      .then((response) => JSON.parse(response[0].result));
```

Reply:
yo tampoco queria usar el self jajaj, el problema es que no estamos llamando a la funcion executeAction aqui como tal, estamos injectando una llamada a la funcion y action_response esta en la extension y no en la tab donde inyectamos el codigo.

Y cuando le pasamos parametros como propones, la tab espera que ese parametro este declarado en el contexto de la tab y ps esta undefined y no le llega nada.

hay algunos workarounds pero creo que es el mejor este